app/src/main/java/nl/eduvpn/app/livedata/ConnectionTimeLiveData.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ *  This file is part of eduVPN.
+ *
+ *     eduVPN is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     eduVPN is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with eduVPN.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package nl.eduvpn.app.livedata
+
+import androidx.annotation.MainThread
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.liveData
+import kotlinx.coroutines.delay
+import nl.eduvpn.app.service.VPNService
+
+/**
+ * Amount of seconds connected to the VPN.
+ */
+object ConnectionTimeLiveData {
```

Comment:
By making these classes objects, you are expanding the scope of these states.
Is this something you want to do deliberately?

Reply:
The idea is that one object can be used by multiple fragments. The unfortunate side effect is indeed that they are not destroyed / garbage collected until the application is closed. Since we are using them in only one fragment anyway I could make it a class and move the creation to the viewmodel to prevent recreation when the fragment is recreated. Not sure what would be best. What would you prefer?

Reply:
I prefer the class version, seems more accurate for what we want.